src/som/compiler/Parser.java

```diff
@@ -1444,6 +1450,29 @@ private LiteralNode literalString() throws ParseError {
     return new StringLiteralNode(s, getSource(coord));
   }
 
+  private LiteralNode literalArray(final MethodBuilder builder) throws ProgramDefinitionError {
+    SourceCoordinate coord = getCoordinate();
+    List<ExpressionNode> expressions = new ArrayList<ExpressionNode>();
+
+    expect(NewLiteralArray, DelimiterOpeningTag.class);
+
+    while (true) {
+      comments();
+
+      switch (sym) {
+        case EndLiteralArray:
+          expect(EndLiteralArray, DelimiterClosingTag.class);
+          return createLiteralArray(expressions, getSource(coord));
+        case Period:
```

Comment:
Yeah, I couldn't come up with a better solution either.

But, I'd prefer to change the naming of the bool, and flip its value, roughly like this:

```Java
boolean needsSeparator = false;
while (true) {
  comments();
  if (sym == EndLiteralArray) {
    expect(EndLiteralArray, DelimiterClosingTag.class);
    return createLiteralArray(expressions, getSource(coord));
  }
  if (needsSeparator) {
    expect(Period, null);
  }
  expressions.add(expression(builder));
  needsSeparator = !accept(Period, StatementSeparatorTag.class);
}
```

When it comes to nicer error messages, you can always use the `expect(.)`/`accept(.)` methods that take a custom message.